deploy/charts/bottlerocket-update-operator/README.md

```diff
@@ -0,0 +1,99 @@
+# Bottlerocket Update Operator Helm Chart
+
+This helmchart carries the templates and resource definitions for
+the bottlerocket-update-operator.
+
+It depends on the `bottlerocket-shadow` CRD chart and should be installed
```

Comment:
Is it possible to model this dependency using Helm itself?

Reply:
> Is it possible to model this dependency using Helm itself?

Good question.

I think [this doc from the helm maintainers on the difficulty handling CRDs ](https://github.com/helm/community/blob/f9e06c16d89ccea1bea77c01a6a96ae3b309f823/architecture/crds.md)is really good prior-art to this PR.

Generally, I believe our posture should be to make few assumptions about the shape of our user's clusters and how they are administered. 

I'll annotate [the following section from the doc](https://github.com/helm/community/blob/f9e06c16d89ccea1bea77c01a6a96ae3b309f823/architecture/crds.md#installing-crds) to give some insight into what I believe makes sense for brupop:

> There are a number of well-described issues with installing CRDs. Users must have broad permissions. A CRD must be fully installed before a manifest that references that CRD can be accepted by the Kubernetes API server. (However, a CR can be accepted before there is anything available to handle the CR event.) It is entirely possible for two different applications to require different versions of the same CRD, but no clear way to support or declare that need within Kubernetes. This is exacerbated by the fact that while a CRD is a global object, the controllers that listen for CRD events may be namespace-bound. This means that two instances of the same controller (in different namespaces) can use the same CRD definition. There is no way to query Kubernetes to discover this fact.

Brupop _does_ have namespaced resources (its API server, controller, daemonsets, etc.) that rely on the global `brs` CRD to be present before install. And while I'd be surprised if anyone was attempting to use `brs` as a dependent resource in their own kubernetes controller, it is ultimately possible. And I believe the really key part of this is "There is no way to query Kubernetes to discover this fact." We can't make assumptions _from the brupop perspective or install_ on if some other namespaced resources are using the `brs` custom resources. Further, we don't want to couple the install of the CRD and the controller too tightly since we do provide multiple versions of the CRD. Therefore, the CRD should be it's own, separate chart.

> All of these things present challenges for installing CRDs. Essentially, any installation process must have some kind of logic like "If the CRD exists, check to make sure that it is compatible with the version of the CRD that I need, and then proceed on my installation. Otherwise, attempt to install the CRD, and handle the case that I might not have permission to do so."

> After looking at several ways of making it easy for chart developers to do this, we alighted on a fairly easy (though inelegant) solution: Helm, as it stands today, provides a separate location in a chart to add CRDs. These will be uploaded before the chart's templates are even rendered (thus avoiding verification issues with the discovery API). If the CRD exists or if the permissions do not allow it, the chart will fail before the application is deployed.

> Of course, users are unhappy with this for a host of reasons. They want CRDs templated (without understanding the race conditions). They want stronger version controls. They don't like having a separate directory for CRDs. We sympathize... but we currently have not devised a better solution for the aforementioned problems.

And unfortunately, this is exactly where we find ourselves; CRDs with a separate chart. Our CRD _does_ need to be templated (in order to derive the namespace and port of the conversion webhook) and we can't rely on pre-rendered templating from the controller chart.